common/flatpak-utils-http.c

```diff
@@ -952,10 +951,13 @@ flatpak_cache_http_uri_once (SoupSession           *soup_session,
 
   if (flags & FLATPAK_HTTP_FLAGS_STORE_COMPRESSED)
     {
+      soup_session_remove_feature_by_type (soup_session, SOUP_TYPE_CONTENT_DECODER);
       soup_message_headers_replace (m->request_headers, "Accept-Encoding",
                                     "gzip");
       data.store_compressed = TRUE;
     }
+  else
+    soup_session_add_feature_by_type (soup_session, SOUP_TYPE_CONTENT_DECODER);
```

Comment:
It comes with this feature by default. Adding multiple can cause issues I believe.

Reply:
Ah good catch, looks like you [only recently fixed that](https://gitlab.gnome.org/GNOME/libsoup/-/commit/14ccc71c7).